spiketoolkit/comparison/sortingcomparison.py

```diff
@@ -205,280 +216,27 @@ def plotConfusionMatrix(self, xlabel=None, ylabel=None):
-        # def confusion_matrix(gtst, sst, pairs, plot_fig=True, xlabel=None, ylabel=None):
-        '''
-
-        Parameters
-        ----------
-        gtst
-        sst
-        pairs 1D array with paired sst to gtst
-
-        Returns
-        -------
-
-        '''
-        sorting1 = self._sorting1
-        sorting2 = self._sorting2
-        unit1_ids = sorting1.getUnitIds()
-        unit2_ids = sorting2.getUnitIds()
-        N1 = len(unit1_ids)
-        N2 = len(unit2_ids)
-
-        conf_matrix = np.zeros((N1 + 1, N2 + 1), dtype=int)
-        idxs_matched = np.where(np.array(self.getMappedSorting1().getMappedUnitIds()) != -1)
-        if len(idxs_matched) > 0:
-            idxs_matched = idxs_matched[0]
-        idxs_unmatched = np.where(np.array(self.getMappedSorting1().getMappedUnitIds()) == -1)
-        if len(idxs_unmatched) > 0:
-            idxs_unmatched = idxs_unmatched[0]
-        unit_map_matched = np.array(self.getMappedSorting1().getMappedUnitIds())[idxs_matched]
-
-        st1_idxs = np.append(np.array(sorting1.getUnitIds())[idxs_matched],
-                             np.array(sorting1.getUnitIds())[idxs_unmatched])
-        st2_matched = unit_map_matched
-        st2_unmatched = []
-
-        for u_i, u1 in enumerate(np.array(sorting1.getUnitIds())[idxs_matched]):
-            lab_st1 = self._labels_st1[u1]
-            tp = len(np.where('TP' == lab_st1)[0])
-            conf_matrix[u_i, u_i] = int(tp)
-            for u_j, u2 in enumerate(sorting2.getUnitIds()):
-                lab_st2 = self._labels_st2[u2]
-                cl_str = str(u1) + '_' + str(u2)
-                cl = len([i for i, v in enumerate(lab_st1) if 'CL' in v and cl_str in v])
-                if cl != 0:
-                    st_p = np.where(u2 == unit_map_matched)
-                    conf_matrix[u_i, st_p] = int(cl)
-            fn = len(np.where('FN' == lab_st1)[0])
-            conf_matrix[u_i, -1] = int(fn)
-
-        for u_i, u1 in enumerate(np.array(sorting1.getUnitIds())[idxs_unmatched]):
-            lab_st1 = self._labels_st1[u1]
-            fn = len(np.where('FN' == lab_st1)[0])
-            conf_matrix[u_i + len(idxs_matched), -1] = int(fn)
-
-        for u_j, u2 in enumerate(sorting2.getUnitIds()):
-            lab_st2 = self._labels_st2[u2]
-            fp = len(np.where('FP' == lab_st2)[0])
-            st_p = np.where(u2 == unit_map_matched)[0]
-            if len(st_p) != 0:
-                conf_matrix[-1, st_p] = int(fp)
-            else:
-                st2_unmatched.append(int(u2))
-                conf_matrix[-1, len(idxs_matched) + len(st2_unmatched) - 1] = int(fp)
-
-        self._confusion_matrix = conf_matrix
-        st2_idxs = np.append(st2_matched, st2_unmatched)
-
+        self._confusion_matrix,  st1_idxs, st2_idxs = do_confusion_matrix(self._sorting1, self._sorting2,
+                                                self._unit_map12, self._labels_st1, self._labels_st2)
+        
         return st1_idxs, st2_idxs
 
+
```

Comment:
Yes I guess this can be moved to comparisontools.

It performs a 1-to-1 spike train match